testflinger_agent/agent.py

```diff
@@ -50,19 +65,43 @@ def _status_worker(self):
                     self.client.post_images(self.advertised_images)
             time.sleep(120)
 
+    def _requests_retry(self, retries=3):
+        session = requests.Session()
+        retry = Retry(
+            total=retries,
+            read=retries,
+            connect=retries,
+            backoff_factor=0.3,
+            status_forcelist=(500, 502, 503, 504),
+        )
+        adapter = HTTPAdapter(max_retries=retry)
+        session.mount("http://", adapter)
+        session.mount("https://", adapter)
+        return session
+
+    def _post_json(self, data):
```

Comment:
Too generic of a name, the agent sends other json data to the server. Maybe post_agent_data() or something like that?